src/pages/Message.jsx

```diff
@@ -1,134 +1,191 @@
 
   useEffect(() => {
-    const unsubscribe = auth.onAuthStateChanged((user) => {
-      setUser(user);
+    console.log("finding user")
+    findUser()
+  }, [userSet]);
+
+  const findUser = async () => {
+    const user = await new Promise((resolve, reject) => {
+      const unsubscribe = auth.onAuthStateChanged((user) => {
+        resolve(user);
+        unsubscribe();
+      }, reject);
     });
-    return unsubscribe;
-  });
 
+    if (!user) {
+      console.log("User not found");
+      return;
+    }
 
-  useEffect(() => {
-    const fetchData = async () => {
-      console.log("new page")
-      try {
-        const collectionRef = collection(firestore, "Child");
-        // get users email from =====>>>>> user.email
-        const q = query(collectionRef, where("email", '==', 'penpalprogram.murphycharity@gmail.com'))
-        const docRef = await getDocs(q)
-        console.log(docRef);
-        if (!docRef.empty) {
-          setChildId(docRef.docs[0].id)
-        }
-      } catch (error) {
-        console.error('Error fetching user:', error);
-      }
-    };
+    const userDocRef = doc(firestore, "users", user.uid);
+    setUser(userDocRef)
+    setUserSet(true);
+  };
 
-    fetchData();
+  useEffect(() => {
+    getSubData()
   }, []);
 
-  useEffect(() => {
-    console.log("new id", childId)
-    let isMounted = true;
-
-    if (!childId) return
-    const fetchData = async () => {
-      try {
-        const collectionRef = collection(firestore, "Chat");
-        const chatBuddy = window.location.pathname.split("/messages/")[1];
-        const intlBuddyRef = doc(collection(firestore, 'InternationalBuddy'), chatBuddy);
-        const childReference = doc(collection(firestore, 'Child'), childId);
-        setChildRef(childReference);
-
-        const q = query(
-          collectionRef,
-          where("child", '==', childReference),
-          where("international_buddy", '==', intlBuddyRef)
-        );
-
-        const docSnapshot = await getDocs(q);
-
-        if (isMounted) {
-          setMessageDocRef(docSnapshot);
-
-          if (!docSnapshot.empty) {
-            setMessage(docSnapshot.docs[0].data().Messages);
-          }
-        }
-      } catch (e) {
-        console.error('Error fetching chat:', e);
+  const collectionName = "letterbox";
+  const splitUrl = window.location.href.split("/")
+  const letterboxId = splitUrl[splitUrl.length - 1]
+
+  const getSubData = async () => {
+    try {
+      const collectionRef = collection(firestore, collectionName);
+      const querySnapshot = await getDocs(query(collectionRef, where("letterboxId", "==", letterboxId)));
+
+      const documentRe = doc(collection(firestore, collectionName), letterboxId);
+      const documentSnapshot = await getDoc(documentRe);
+
+      const subcollectionRe = collection(documentRe, "letters");
+      console.log("collecting messages")
+      const q = query(subcollectionRe, orderBy("created_at", "desc"), limit(1));
```

Comment:
How does this work? Is it really at a `created_at` timestamp? Also, please paginate by 10 messages.

Reply:
Cuz I don't see the field in the documents.

Reply:
created_at is in the subcollection documents